src/hotspot/share/ci/ciEnv.cpp

```diff
@@ -1499,21 +1497,20 @@ void ciEnv::record_call_site_method(Thread* thread, Method* adapter) {
 
 // Process an invokedynamic call site and record any dynamic locations.
 void ciEnv::process_invokedynamic(const constantPoolHandle &cp, int indy_index, JavaThread* thread) {
-  int index = cp->decode_invokedynamic_index(indy_index);
-  ResolvedIndyEntry* indy_info = cp->resolved_indy_entry_at(index);
+  ResolvedIndyEntry* indy_info = cp->resolved_indy_entry_at(indy_index);
   if (indy_info->method() != nullptr) {
     // process the adapter
     Method* adapter = indy_info->method();
     record_call_site_method(thread, adapter);
     // process the appendix
-    oop appendix = cp->resolved_reference_from_indy(index);
+    oop appendix = cp->resolved_reference_from_indy(indy_index);
     {
       RecordLocation fp(this, "<appendix>");
       record_call_site_obj(thread, appendix);
     }
     // process the BSM
     int pool_index = indy_info->constant_pool_index();
-    BootstrapInfo bootstrap_specifier(cp, pool_index, index);
+    BootstrapInfo bootstrap_specifier(cp, pool_index, indy_index);
```

Comment:
Why not just change the incoming parameter name to `index`?

Reply:
`indy_index` is used frequently even in functions that are only used in the context of invokedynamic. I think it helps with clarity.